### PR TITLE
Fix dark mode status #158

### PR DESCRIPTION
**Pull request by metinbaris**
## Description

Gets dark mode status from local storage, sets body class whenever the app.js is loaded

### CODE REVIEW

**Comment by SallyMcGrath** [comment]
Nice!

Do you think you have capacity to turn this into a miniature web-component instead of throwing it in app.js?  OK if not - I can make a new ticket.

**Comment by metinbaris** [comment]
> Nice!
> 
> Do you think you have capacity to turn this into a miniature web-component instead of throwing it in app.js? OK if not - I can make a new ticket.

@SallyMcGrath ahh ok just tried what you mentioned, sounds good to me :)

File: assets/scripts/dark-mode.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    const isDarkModeOn = localStorage.getItem("isDarkMode");
+    if (isDarkModeOn === "1") {
+      localStorage.removeItem("isDarkMode");
+    } else {
+      localStorage.setItem("isDarkMode", "1");
+    }
+  }
+
+  render() {
+    this.shadowRoot.innerHTML = `<button id="mode-toggle" type="button"
```
Comment (SallyMcGrath) [inline assets/scripts/dark-mode.js:34]: There's a simpler way to do this. You can project your button in from the Light DOM using a slot render. 

```
render() {
    this.shadowRoot.innerHTML = `<slot name="button"></slot>`;
  }
```

And then in foot.html, return the html with these 2 small changes

```html
  <dark-mode class="l-footer__mode"
    ><button
      slot="button"
      id="mode-toggle"
      type="button"
      class="e__button--toggle e-button--icon">
      <svg
        xmlns="http://www.w3.org/2000/svg"
        fill="none"
        viewBox="0 0 24 24"
        stroke-width="1.5"
        stroke="currentColor"
        class="e-icon e__button__icon e-icon--moon">
        <path
          stroke-linecap="round"
          stroke-linejoin="round"
          d="M21.752 15.002A9.718 9.718 0 0118 15.75c-5.385 0-9.75-4.365-9.75-9.75 0-1.33.266-2.597.748-3.752A9.753 9.753 0 003 11.25C3 16.635 7.365 21 12.75 21a9.753 9.753 0 009.002-5.998z" />
      </svg>
      <span class="is-invisible">Toggle between light and dark mode</span>
    </button></dark-mode
  >
```

Comment (SallyMcGrath) [reply assets/scripts/dark-mode.js:34]: eg put the positioning class on the web component and the slot attribute on the button

File: assets/scripts/dark-mode.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    super();
+    this.attachShadow({ mode: "open" });
+  }
+
+  connectedCallback() {
+    this.render();
+    this.setDarkMode();
+  }
+
+  setDarkMode() {
```
Comment (SallyMcGrath) [inline assets/scripts/dark-mode.js:12]: I think there's a slightly simpler way to do this too. How about something more like

```js
toggleDarkMode = () => {
    this.isDarkMode = !this.isDarkMode;
    localStorage.setItem("isDarkMode", this.isDarkMode ? "1" : "0");
    this.updateBodyClass();
  };

  updateBodyClass() {
    document.body.classList.toggle("is-dark-mode", this.isDarkMode);
    document.body.classList.toggle("is-light-mode", !this.isDarkMode);
  }
```

File: assets/scripts/dark-mode.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      stroke="currentColor"
+      class="e-icon e__button__icon e-icon--moon">
+      <path stroke-linecap="round" stroke-linejoin="round"
+      d="M21.752 15.002A9.718 9.718 0 0118 15.75c-5.385 0-9.75-4.365-9.75-9.75 0-1.33.266-2.597.748-3.752A9.753 9.753 0 003 11.25C3 16.635 7.365 21 12.75 21a9.753 9.753 0 009.002-5.998z" />
+    </svg>
+    <span class="is-invisible">Toggle between light and dark mode</span>
+    </button>`;
+
+    // Replace the host element with its shadow content
+    const hostElement = document.querySelector('dark-mode');
```
Comment (SallyMcGrath) [inline assets/scripts/dark-mode.js:50]: I think you can remove all of this and just project your existing html in. You don't need to engage with outerHTML.

**Review by SallyMcGrath — CHANGES_REQUESTED** [review]
Nice! Beautiful work!

I've made a few comments where you could make this a bit simpler and use some of the fun powers of web components. 

Here's a rough draft putting that together, if it's helpful. Just a bit simpler and touching the DOM a little less. Take a look and see if you can use any of it.

```js
class DarkMode extends HTMLElement {
  constructor() {
    super();
    this.attachShadow({ mode: "open" });
    this.isDarkMode = localStorage.getItem("isDarkMode") === "1";
    this.toggleButton = null;
  }

  connectedCallback() {
    this.render();
    this.updateBodyClass();
    this.attachListener();
  }

  toggleDarkMode = () => {
    this.isDarkMode = !this.isDarkMode;
    localStorage.setItem("isDarkMode", this.isDarkMode ? "1" : "0");
    this.updateBodyClass();
  };

  updateBodyClass() {
    document.body.classList.toggle("is-dark-mode", this.isDarkMode);
    document.body.classList.toggle("is-light-mode", !this.isDarkMode);
  }

  attachListener() {
    this.toggleButton = this.querySelector('[slot="button"]');
    if (this.toggleButton) {
      this.toggleButton.addEventListener("click", this.toggleDarkMode);
    }
  }

  render() {
    this.shadowRoot.innerHTML = `<slot name="button"></slot>`;
  }
}

customElements.define("dark-mode", DarkMode);
```

File: assets/scripts/dark-mode.js (hunk 3 above)
Comment (metinbaris) [reply assets/scripts/dark-mode.js]: @SallyMcGrath This was about styling, and it seems the slot element works perfectly fine in these kinds of cases.

**Review by SallyMcGrath — APPROVED** [review]
Sweet! You're a star ⭐

**Pull request merged**